DllImportGenerator/DllImportGenerator/TypePositionInfo.cs

```diff
@@ -208,15 +208,29 @@ static MarshalAsInfo CreateMarshalAsInfo(AttributeData attrData, DefaultMarshall
                     }
                 }
 
-                return isArrayType
-                    ? new ArrayMarshalAsInfo(
+                if (isArrayType)
```

Comment:
Seems like we are at the end of the method here (i.e. last logical conclusions). Let's flip this predicate and return here in the non-array case and remove the nesting for the more complicated follow-up to handle arrays.